yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -702,4 +702,15 @@ export class KVPxeDatabase implements PxeDatabase {
   #getTaggingSecretsIndexes(appTaggingSecrets: Fr[], storageMap: AztecMap<string, number>): Promise<number[]> {
     return this.db.transaction(() => appTaggingSecrets.map(secret => storageMap.get(`${secret.toString()}`) ?? 0));
   }
+
+  async resetNoteSyncData(): Promise<void> {
+    await this.db.transaction(() => {
+      for (const recipient of this.#taggingSecretIndexesForRecipients.keys()) {
+        void this.#taggingSecretIndexesForRecipients.delete(recipient);
+      }
+      for (const sender of this.#taggingSecretIndexesForSenders.keys()) {
```

Comment:
This is bad until https://github.com/AztecProtocol/aztec-packages/pull/10071 gets in, since it would lead to rampant index reuse